intellij/src/saros/intellij/eventhandler/filesystem/LocalFilesystemModificationHandler.java

```diff
@@ -617,11 +630,22 @@ private void generateFileMove(
     SPath oldFilePath = VirtualFileConverter.convertToSPath(project, oldFile);
     SPath newParentPath = VirtualFileConverter.convertToSPath(project, newBaseParent);
 
+    IResource oldResource =
+        intelliJReferencePointManager.getSarosResource(
+            oldFilePath.getReferencePoint(), oldFilePath.getProjectRelativePath());
+
+    IFile oldSarosFile = oldResource.adaptTo(IntelliJFileImpl.class);
+
+    IResource newResource =
+        intelliJReferencePointManager.getSarosResource(
+            newParentPath.getReferencePoint(), newParentPath.getProjectRelativePath());
+
+    IFile newSarosParent = newResource.adaptTo(IntelliJFileImpl.class);
```

Comment:
This will always be a folder and not a file.

Why do you need to resolve the resource type here anyway? (Same goes for the `oldResource`.)

Reply:
Why should this always be a folder??

The `IntelliJReferencePointManager` only provides resolving the Saros resource for a given IReferencePoint and relative path. So the `IntelliJReferencePointManager` can be extended, so that it also returns a Saros File and/or folder. Or the IntelliJReferencePointManager resolves the Saros resources as `IResource` and converts it to IFile or IFolder, if it is really needed.

Reply:
> Why should this always be a folder??

This is always a folder as we don't have a reference for the new resource yet (as it does not exist). Instead, we use the reference for the *parent* of the new resource, which is why the SPath was called `newParentPath`. As it is a parent resource, it *must* be a folder.

So, in the new implementation, this should also be called `newParentResource`. The actual new resource reference is created in l.687 and l.716.

Reply:
Ok, thank you for your explanation. Now I know what you meant. I adjusted the code fragment.

Reply:
I would merge this PR, if you agree with the changes